File: algoritmos/cesar.cs
using System;

public class Test
{
    static string simbolos = "ABCDEFGHIJKLMNOPQRSTUVWXYZ,_.?";
    
	public static void Main()
	{
		//Se realiza el cifrado cesar con +25 mod 30 como unidad de desplazamiento, la llave para descifrar es +5 mod 30 
		string mensaje = ",P?IJNW.DZNWZWOJ.JN";
		Console.WriteLine(decodificarMensaje(mensaje));
	}
	
	public static string decodificarMensaje(string s){
        string decoded_message = "";
        for (int i = 0; i < s.Length; i++) {
            /*int indice = Math.floorMod(simbolos.indexOf(s.charAt(i)) * 13 + 21, 30);*/
            int indice = (simbolos.IndexOf(s[i]) + 5) % 30;
            decoded_message += simbolos[indice];
        }
        return decoded_message;
    }
}

File: algoritmos/cesar.java
import java.util.*;
import java.lang.*;
import java.io.*;

/* Name of the class has to be "Main" only if the class is public. */
class Codechef
{
    //Se realiza el cifrado cesar con +25 mod 30 como unidad de desplazamiento, la llave para descifrar es +5 mod 30 
    static String simbolos = "ABCDEFGHIJKLMNOPQRSTUVWXYZ,_.?";

    public static void main(String[] args) {
        String mensaje = ",P?IJNW.DZNWZWOJ.JN";
        System.out.println(decodificarMensaje(mensaje));
    }
    
     public static String decodificarMensaje(String s){
        String decoded_message = "";
        for (int i = 0; i < s.length(); i++) {
            int indice = (simbolos.indexOf(s.charAt(i))+5) % 30;
            decoded_message += simbolos.charAt(indice);
        }
        return decoded_message;
    }
}
